tests/scripts/check_files.py

```diff
@@ -36,7 +36,7 @@
 
 import scripts_path # pylint: disable=unused-import
 from mbedtls_dev import build_tree
-
+from detect_unicode_chars import load_and_check_file as unicode_check_f
```

Comment:
This looks complicated. We don't use a lot of Unicode. Can't we make a simple whitelist of permitted byte sequences or characters?

Reply:
Agreed. Also, reading in as unicode could fail if there are invalid byte sequences - these should be reported (i.e. "invalid byte sequence at line NN")